opentelemetry-sdk/src/trace/span.rs

```diff
@@ -246,6 +256,282 @@ impl Drop for Span {
     }
 }
 
+/// Represents a finished span passed to a span processor.
+///
+/// The data associated with the span is not writable, but it can be read
+/// through the `ReadableSpan` trait.
+///
+/// Taking ownership of the span data is done by calling `consume`.
+/// If `consume`` is never called, the on_ending method will not perform any copy of
+/// the span data.
+///
+/// Calling any `ReadableSpan` method on the `FinishedSpan` will panic if the span data
```

Comment:
We should probably error out through the internal logging macros here and not panic. We try to avoid panic-ing wherever we can.

Reply:
This does not panic anymore